ca/server.go

```diff
@@ -32,6 +34,45 @@ type APISecurityConfigUpdater interface {
 
 var _ APISecurityConfigUpdater = &Server{}
 
+// ExternalCATLSUpdater is a way to notify the server of TLS updates for the external CA
+type ExternalCATLSUpdater interface {
+	NotifyTLSChange() <-chan events.Event
+	GetCertificates() []tls.Certificate
+}
+
+// SecurityConfigTLSUpdater takes a security config and sets a watch so we can tell when TLS credentials have changed
+type SecurityConfigTLSUpdater struct {
```

Comment:
I can pass a channel instead to the CA server notifying it when the TLS creds have been updated instead.  But the `SecurityConfig` object already had a watch on it to notify the agent when the credentials or the root CA have changed so the agent session could restart.  It seemed easier to bottleneck it through the `SecurityConfig`, but I could also instead try setting a private watch in the `node.go` code.